build/v1/types.go

```diff
@@ -1335,3 +1348,59 @@ type SecretSpec struct {
 	// mountPath is the path at which to mount the secret
 	MountPath string `json:"mountPath" protobuf:"bytes,2,opt,name=mountPath"`
 }
+
+// BuildVolume describes a volume that is made available to build pods,
+// such that it can be mounted into buildah's runtime environment.
+// Only a subset of Kubernetes Volume sources are supported.
+type BuildVolume struct {
+	// name is a unique identifier for this BuildVolume.
+	// It must conform to the Kubernetes DNS label standard and be unique within the pod.
+	// It cannot collide with volumes that are always added by the build controller.
+	// More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names
+	Name string `json:"name" protobuf:"bytes,1,opt,name=name"`
+
+	// source represents the location and type of the mounted volume.
+	Source BuildVolumeSource `json:"source" protobuf:"bytes,2,opt,name=source"`
+
+	// mounts represents the location and read/write status of the volume
+	// in the buildah container
+	Mounts []BuildVolumeMount `json:"mounts" protobuf:"bytes,3,opt,name=mounts"`
```

Comment:
did we actually confirm that a single path can be mounted inside the buildah container multiple times?

Reply:
I am launching a cluster to verify and will report back.

Reply:
Yes, a single path can be mounted inside the buildah container multiple times:
```
volumes:
      - mounts:
        - destinationPath: /var/run/secrets/some-secret
        - destinationPath: /var/run/secrets/some-secret-2
        name: this-secret
        source:
          secret:
            defaultMode: 420
            secretName: my-secret
          type: Secret
```
```
STEP 11: RUN /bin/sh -ic 'cat /var/run/secrets/some-secret/key; cat /var/run/secrets/some-secret-2/key'
my-secret-value my-secret-value
```